### PR TITLE
nodes/tryUntil: add argument for Sleep inside the for loop

### PR DESCRIPTION
**Pull request by neolit123**
/kind cleanup
can adjust sleep times if needed.

### CODE REVIEW

File: pkg/cluster/nodes/node.go [hunk 1]
```diff
@@ -102,7 +102,7 @@ func (n *Node) CopyFrom(source, dest string) error {
 // WaitForDocker waits for Docker to be ready on the node
 // it returns true on success, and false on a timeout
 func (n *Node) WaitForDocker(until time.Time) bool {
-	return tryUntil(until, func() bool {
+	return tryUntil(until, time.Second, func() bool {
```
Comment (BenTheElder) [inline pkg/cluster/nodes/node.go]: hmm, how did we pick these values?

the downside to this is increasing the latency from ready -> cluster up progressing
these commands are somewhat self throttling by taking time themselves.

Comment (neolit123) [reply pkg/cluster/nodes/node.go]: > hmm, how did we pick these values?

experiment from yesterday.

> the downside to this is increasing the latency from ready -> cluster up progressing
these commands are somewhat self throttling by taking time themselves.

it executes something like a 100 kubectl calls until it gets the signal that the CP is ready.

Comment (BenTheElder) [reply pkg/cluster/nodes/node.go]: > it executes something like a 100 kubectl calls until it gets the signal that the CP is ready.

sounds working as intended 😉 

the docker call should be much more throttled, this one is much slower

Comment (tao12345666333) [reply pkg/cluster/nodes/node.go]: SGTM.  :+1  Do we need to do more experiments on different performance machines? I have some local machines available for testing.

**Comment by BenTheElder** [comment]
FWIW I do think this probably makes sense, but I'd like to use a more rigorous value and possibly not throttle waiting for docker 🤔

**Pull request closed without merging**